src/printf/iter_tools.rs

```diff
@@ -0,0 +1,102 @@
+
+/*
+  This module uses code from the iter-tools
+  project, licensed here under the MIT license
```

Comment:
A link to the project would be nice. Also it's probably best to paste the iter-tools license here.